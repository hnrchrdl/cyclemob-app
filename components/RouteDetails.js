import React from 'react';
import { StyleSheet, View, Text } from 'react-native';
import PropTypes from 'prop-types';
import Button from './Button';
import { sToTime, mToKm } from '../lib/helper';

RouteDetails.propTypes = {
  distance: PropTypes.number.isRequired,
  duration: PropTypes.number.isRequired,
  onRemove: PropTypes.func.isRequired,
  onClose: PropTypes.func.isRequired
};

function RouteDetails({ distance, duration, onRemove, onClose }) {
  return (
    <View style={styles.container}>
      <View style={styles.containerInner}>
        <Text>{mToKm(distance)}</Text>
        <Text>{sToTime(duration)}</Text>
        <Button iconName="close" onPress={onClose} />
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    padding: 10,
    backgroundColor: 'rgba(255,255,255,0.8)'
  },
  containerInner: {
    flexDirection: 'row',
    justifyContent: 'space-between'
  }
});

export default RouteDetails;
