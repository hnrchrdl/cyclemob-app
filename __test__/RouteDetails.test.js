import React from 'react';
import { shallow } from 'enzyme';
import RouteDetails from '../components/RouteDetails';

const props = {
  distance: 100,
  duration: 40,
  onClose: jest.fn(),
  onRemove: jest.fn()
};

it('renders without crashing', () => {
  const tree = shallow(<RouteDetails {...props} />);
  expect(tree).toBeTruthy();
});

it('matches snapshot', () => {
  const tree = shallow(<RouteDetails {...props} />);
  expect(tree).toMatchSnapshot();
});

it('reacts to marker changes', () => {
  const tree = shallow(<RouteDetails {...props} />);
  tree.setProps({
    distance: 150,
    duration: 10
  });
  expect(tree).toMatchSnapshot();
});

it('handles close button presses', () => {
  const tree = shallow(<RouteDetails {...props} />);
  tree
    .find('Button')
    .find({ iconName: 'close' })
    .simulate('press');
  expect(props.onClose).toHaveBeenCalledTimes(1);
});

// it('handles remove button presses', () => {
//     const tree = shallow(<RouteDetails {...props} />);
//     tree
//       .find('Button')
//       .find({ iconName: 'close' })
//       .simulate('press');
//     expect(props.onClose).toHaveBeenCalledTimes(1);
//   });
