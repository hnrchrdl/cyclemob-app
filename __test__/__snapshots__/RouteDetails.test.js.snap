// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RouteDetails
    distance={100}
    duration={40}
    onClose={[MockFunction]}
    onRemove={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Component
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          0.10 km
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          00:00:40
        </Text>
        <Button
          iconName="close"
          onPress={[MockFunction]}
        />
      </Component>,
      "style": Object {
        "backgroundColor": "rgba(255,255,255,0.8)",
        "padding": 10,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            0.10 km
          </Text>,
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            00:00:40
          </Text>,
          <Button
            iconName="close"
            onPress={[MockFunction]}
          />,
        ],
        "style": Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "0.10 km",
            "ellipsizeMode": "tail",
          },
          "ref": null,
          "rendered": "0.10 km",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "00:00:40",
            "ellipsizeMode": "tail",
          },
          "ref": null,
          "rendered": "00:00:40",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "iconName": "close",
            "onPress": [MockFunction],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Component
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            0.10 km
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            00:00:40
          </Text>
          <Button
            iconName="close"
            onPress={[MockFunction]}
          />
        </Component>,
        "style": Object {
          "backgroundColor": "rgba(255,255,255,0.8)",
          "padding": 10,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            >
              0.10 km
            </Text>,
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            >
              00:00:40
            </Text>,
            <Button
              iconName="close"
              onPress={[MockFunction]}
            />,
          ],
          "style": Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "0.10 km",
              "ellipsizeMode": "tail",
            },
            "ref": null,
            "rendered": "0.10 km",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "00:00:40",
              "ellipsizeMode": "tail",
            },
            "ref": null,
            "rendered": "00:00:40",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "iconName": "close",
              "onPress": [MockFunction],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`reacts to marker changes 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RouteDetails
    distance={150}
    duration={10}
    onClose={[MockFunction]}
    onRemove={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Component
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          0.15 km
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          00:00:10
        </Text>
        <Button
          iconName="close"
          onPress={[MockFunction]}
        />
      </Component>,
      "style": Object {
        "backgroundColor": "rgba(255,255,255,0.8)",
        "padding": 10,
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            0.15 km
          </Text>,
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            00:00:10
          </Text>,
          <Button
            iconName="close"
            onPress={[MockFunction]}
          />,
        ],
        "style": Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "0.15 km",
            "ellipsizeMode": "tail",
          },
          "ref": null,
          "rendered": "0.15 km",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "00:00:10",
            "ellipsizeMode": "tail",
          },
          "ref": null,
          "rendered": "00:00:10",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "iconName": "close",
            "onPress": [MockFunction],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Component
          style={
            Object {
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            0.15 km
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            00:00:10
          </Text>
          <Button
            iconName="close"
            onPress={[MockFunction]}
          />
        </Component>,
        "style": Object {
          "backgroundColor": "rgba(255,255,255,0.8)",
          "padding": 10,
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            >
              0.15 km
            </Text>,
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            >
              00:00:10
            </Text>,
            <Button
              iconName="close"
              onPress={[MockFunction]}
            />,
          ],
          "style": Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "0.15 km",
              "ellipsizeMode": "tail",
            },
            "ref": null,
            "rendered": "0.15 km",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "00:00:10",
              "ellipsizeMode": "tail",
            },
            "ref": null,
            "rendered": "00:00:10",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "iconName": "close",
              "onPress": [MockFunction],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
